node/src/accountant/scanners/payable_scanner/utils.rs

```diff
@@ -0,0 +1,565 @@
+// Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::accountant::comma_joined_stringifiable;
+use crate::accountant::db_access_objects::failed_payable_dao::{FailedTx, FailureStatus};
+use crate::accountant::db_access_objects::payable_dao::{PayableAccount, PayableDaoError};
+use crate::accountant::db_access_objects::pending_payable_dao::PendingPayable;
+use crate::accountant::db_access_objects::sent_payable_dao::Tx;
+use crate::accountant::db_access_objects::utils::{ThresholdUtils, TxHash};
+use crate::accountant::db_access_objects::Transaction;
+use crate::accountant::scanners::payable_scanner::msgs::InitialTemplatesMessage;
+use crate::accountant::scanners::payable_scanner::tx_templates::initial::new::NewTxTemplates;
+use crate::accountant::scanners::payable_scanner::tx_templates::initial::retry::RetryTxTemplates;
+use crate::blockchain::blockchain_interface::data_structures::BatchResults;
+use crate::sub_lib::accountant::PaymentThresholds;
+use crate::sub_lib::wallet::Wallet;
+use bytes::Buf;
+use itertools::{Either, Itertools};
+use masq_lib::logger::Logger;
+use masq_lib::ui_gateway::NodeToUiMessage;
+use std::cmp::Ordering;
+use std::collections::{BTreeSet, HashMap};
+use std::ops::Not;
+use std::time::SystemTime;
+use thousands::Separable;
+use web3::types::{Address, H256};
+
+#[derive(Debug, PartialEq)]
+pub struct PayableScanResult {
+    pub ui_response_opt: Option<NodeToUiMessage>,
+    pub result: NextScanToRun,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum NextScanToRun {
+    PendingPayableScan,
+    NewPayableScan,
+    RetryPayableScan,
+}
+
+pub fn filter_receiver_addresses_from_txs<'a, T, I>(transactions: I) -> BTreeSet<Address>
+where
+    T: 'a + Transaction,
```

Comment:
I see...
Because of this you have come up with the Transaction trait. Uhhh. Worth it? 

Then I'd say you can keep it but remove all the unneeded methods.

Reply:
They might be useful in the future. I'd say let's keep them, they are like the getter methods.